File: papers/miniKanren/prolog/app/original.pl
appendo(nil, Y, Y).
appendo(cons(H, T), Y, cons(H, Ty)) :- appendo(T, Y, Ty).

?- appendo(X,Y,cons(1,cons(2,cons(3,cons(4,cons(5,cons(6,cons(7,cons(8,cons(9,cons(10,cons(11,cons(12,cons(13,cons(14,cons(15,nil)))))))))))))))).

File: papers/miniKanren/ocanren/da/original.ml
open GT
open OCanren
open OCanren.Std
open Helper

let topLevel x0 x1 x2 x3 = let rec doubleAppendo x y z r = (fresh (t) (((appendo x y t) &&& (appendo t z r)))) and appendo x y xy = (((x === (Std.List.nil ())) &&& (xy === y)) ||| (fresh (ty t h) (((x === (Std.(%) (h) (t))) &&& ((xy === (Std.(%) (h) (ty))) &&& (appendo t y ty)))))) in (doubleAppendo x0 x1 x2 x3)
